enginetest/query_plans.go

```diff
@@ -244,18 +244,16 @@ var PlanTests = []QueryPlanTest{
 	{
 		Query: `SELECT sub.i, sub.i2, sub.s2, ot.i2, ot.s2 FROM othertable ot LEFT JOIN (SELECT i, i2, s2 FROM mytable INNER JOIN othertable ON i = i2 WHERE CONVERT(s2, signed) <> 0) sub ON sub.i = ot.i2 WHERE ot.i2 > 0`,
 		ExpectedPlan: "Project(sub.i, sub.i2, sub.s2, ot.i2, ot.s2)\n" +
-			" └─ LeftJoin(sub.i = ot.i2)\n" +
-			"     ├─ Filter(ot.i2 > 0)\n" +
-			"     │   └─ TableAlias(ot)\n" +
-			"     │       └─ IndexedTableAccess(othertable on [othertable.i2] with ranges: [{(0, ∞)}])\n" +
-			"     └─ HashLookup(child: (sub.i), lookup: (ot.i2))\n" +
-			"         └─ CachedResults\n" +
-			"             └─ SubqueryAlias(sub)\n" +
-			"                 └─ Project(mytable.i, othertable.i2, othertable.s2)\n" +
-			"                     └─ IndexedJoin(mytable.i = othertable.i2)\n" +
-			"                         ├─ Table(mytable)\n" +
-			"                         └─ Filter(NOT((convert(othertable.s2, signed) = 0)))\n" +
-			"                             └─ IndexedTableAccess(othertable on [othertable.i2])\n" +
+			" └─ Filter(ot.i2 > 0)\n" +
+			"     └─ LeftIndexedJoin(sub.i = ot.i2)\n" +
```

Comment:
This looks incorrect to me. `ot` needs to be the primary table in this join or the semantics are incorrect. Right now this includes every row from the subquery and returns matching rows from `ot`, and it should be the opposite.

Reply:
Are the correctness tests not thorough enough? I would need to spend a few more days taking notes on all of our join code to give a good answer.